### PR TITLE
Chart sort dialog supports date attributes

### PR DESCRIPTION
**Pull request by xMort**
Some miscellaneous issues from TNT-513 were fixed as well.

JIRA: TNT-461, TNT-513

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by xMort** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7089/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8942/ : FAILURE in 9m 36s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7090/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1396/ : SUCCESS in 18m 30s

**Comment by xMort** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7092/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1397/ : SUCCESS in 19m 43s

**Comment by xMort** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1398/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by xMort** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7099/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8947/ : SUCCESS in 13m 59s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1399/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7100/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1399/ : UNSTABLE in 23m 33s

**Comment by xMort** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7101/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8948/ : SUCCESS in 16m 36s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1400/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7102/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1400/ : UNSTABLE in 21m 25s

**Comment by xMort** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7103/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8949/ : SUCCESS in 13m 51s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1401/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7104/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/1401/ : UNSTABLE in 21m 35s

File: libs/sdk-ui-kit/api/sdk-ui-kit.api.md [hunk 1]
```diff
@@ -1560,6 +1557,16 @@ export interface IHubspotFormValue {
     [key: string]: string | number | boolean;
 }
 
+// @internal (undocumented)
+export interface IIBucketItem {
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-kit/api/sdk-ui-kit.api.md:1561]: Quite strange name of interface

File: libs/sdk-ui-kit/src/ChartSorting/AttributeDropdown/AttributeDropdown.tsx [hunk 2]
```diff
@@ -80,18 +104,67 @@ const getSortTypeItems = (available: IAvailableSortsGroup, intl: IntlShape): ISo
             },
         );
     }
-
+    if (available.attributeSort.normalSortEnabled && bucketItem.type === "genericDate") {
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-kit/src/ChartSorting/AttributeDropdown/AttributeDropdown.tsx:107]: would not be more readable to keep all variants of normal sorting together?
if (available.attributeSort.normalSortEnabled)
{
 if(attribute) push
 if(chrono) push
 if(generic) push
}

File: libs/sdk-ui-kit/src/ChartSorting/test/ChartSorting.test.tsx [hunk 3]
```diff
@@ -140,10 +184,22 @@ describe("ChartSorting", () => {
     });
 
     describe("onApply", () => {
+        it("should not call onApply when disabled Apply button is clicked", () => {
+            const onApply = jest.fn();
+            const component = renderComponent({ onApply });
+            clickApplyButton(component);
+
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-kit/src/ChartSorting/test/ChartSorting.test.tsx:191]: if it is easy what about to check disabled state/class of button too?

File: libs/sdk-ui-kit/src/ChartSorting/AttributeDropdown/AttributeDropdown.tsx (hunk 2 above)
Comment (xMort) [reply libs/sdk-ui-kit/src/ChartSorting/AttributeDropdown/AttributeDropdown.tsx:107]: I had it like that originally but I needed to be able to have `Default` option as the last one below numerical sort as in the mockups and this was the easiest way.

image

File: libs/sdk-ui-kit/api/sdk-ui-kit.api.md (hunk 1 above)
Comment (xMort) [reply libs/sdk-ui-kit/api/sdk-ui-kit.api.md]: Fixed.

File: libs/sdk-ui-kit/src/ChartSorting/test/ChartSorting.test.tsx (hunk 3 above)
Comment (xMort) [reply libs/sdk-ui-kit/src/ChartSorting/test/ChartSorting.test.tsx:192]: Fixed.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7109/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8952/ : SUCCESS in 18m 54s

File: libs/sdk-ui-kit/src/ChartSorting/AttributeDropdown/AttributeDropdown.tsx (hunk 2 above)
Comment (ivan-nejezchleb) [reply libs/sdk-ui-kit/src/ChartSorting/AttributeDropdown/AttributeDropdown.tsx:107]: ok

**Review by ivan-nejezchleb — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2305

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2053/ : SUCCESS in 1m 06s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2336/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1150/ : SUCCESS in 14m 32s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8954/ : SUCCESS in 16m 31s

**Pull request merged**